src/Kernel/Context.class.st

```diff
@@ -2251,6 +2262,16 @@ Context >> unwindTo: aContext [
 			unwindBlock value ]].
 ]
 
+{ #category : #'debugger access' }
+Context >> variableNamesOf: anObject [
+
+	"Retrieve the names of the arguments and temporary variables, if any, that refer to the given object."
+
+	^self astScope allTemps
+		select: [ :v | (v readInContext: self) == anObject ]
+		thenCollect: [ :v | v name ]
```

Comment:
Why you want just names? We should stop think in terms of primitive objects. Just return full objects representing temp variables.
I would also rename the method to #tempVariablesContaining: . Then you can also define allVariablesContaining: which will lookup the receiver in addition.